pkg/schedule/schedulers/evict_leader.go

```diff
@@ -163,6 +153,109 @@ func (conf *evictLeaderSchedulerConfig) getKeyRangesByID(id uint64) []core.KeyRa
 	return nil
 }
 
+func (conf *evictLeaderSchedulerConfig) encodeConfig() ([]byte, error) {
+	conf.RLock()
+	defer conf.RUnlock()
+	return EncodeConfig(conf)
+}
+
+func (conf *evictLeaderSchedulerConfig) reloadConfig(name string) error {
+	conf.Lock()
+	defer conf.Unlock()
+	cfgData, err := conf.storage.LoadSchedulerConfig(name)
+	if err != nil {
+		return err
+	}
+	if len(cfgData) == 0 {
+		return nil
+	}
+	newCfg := &evictLeaderSchedulerConfig{}
+	if err = DecodeConfig([]byte(cfgData), newCfg); err != nil {
+		return err
+	}
+	pauseAndResumeLeaderTransfer(conf.cluster, conf.StoreIDWithRanges, newCfg.StoreIDWithRanges)
+	conf.StoreIDWithRanges = newCfg.StoreIDWithRanges
+	conf.Batch = newCfg.Batch
+	return nil
+}
+
+func (conf *evictLeaderSchedulerConfig) pauseLeaderTransfer(cluster sche.SchedulerCluster) error {
+	conf.RLock()
+	defer conf.RUnlock()
+	var res error
+	for id := range conf.StoreIDWithRanges {
+		if err := cluster.PauseLeaderTransfer(id); err != nil {
+			res = err
+		}
+	}
+	return res
+}
+
+func (conf *evictLeaderSchedulerConfig) resumeLeaderTransfer(cluster sche.SchedulerCluster) {
+	conf.RLock()
+	defer conf.RUnlock()
+	for id := range conf.StoreIDWithRanges {
+		cluster.ResumeLeaderTransfer(id)
+	}
+}
+
+func (conf *evictLeaderSchedulerConfig) pauseLeaderTransferIfStoreNotExist(id uint64) (bool, error) {
+	conf.RLock()
+	defer conf.RUnlock()
+	if _, exist := conf.StoreIDWithRanges[id]; !exist {
+		if err := conf.cluster.PauseLeaderTransfer(id); err != nil {
+			return exist, err
+		}
+	}
+	return true, nil
+}
+
+func (conf *evictLeaderSchedulerConfig) update(id uint64, newRanges []core.KeyRange, batch int) error {
+	conf.Lock()
+	defer conf.Unlock()
+	if id != 0 {
+		conf.StoreIDWithRanges[id] = newRanges
+	}
+	conf.Batch = batch
+	err := conf.persistLocked()
+	if err != nil {
+		conf.removeStoreLocked(id)
+	}
+	return err
+}
+
+func (conf *evictLeaderSchedulerConfig) delete(id uint64) (any, error) {
+	conf.Lock()
+	keyRanges := conf.StoreIDWithRanges[id]
+	succ, last := conf.removeStoreLocked(id)
+	var resp any
+
+	if !succ {
+		conf.Unlock()
+		return resp, errs.ErrScheduleConfigNotExist.FastGenByArgs()
+	}
+
+	err := conf.persistLocked()
+	if err != nil {
+		conf.resetStoreLocked(id, keyRanges)
+		conf.Unlock()
+		return resp, err
+	}
+	if !last {
+		conf.Unlock()
+		return resp, nil
+	}
+	conf.Unlock()
+	if err := conf.removeSchedulerCb(EvictLeaderName); err != nil {
```

Comment:
Is it better to `removeSchedulerCb` inside the `removeStoreLocked`? Is it a heavy operation that you don't want to put it into a xxxxLocked function? (But I think updateConfig/deleteConfig are low frequency operations 🤔)

Reply:
No, there is a lock problem, if we put it into removeStoreLocked, it might have a deadlock issue.

Reply:
Sure, you are right. `removeSchedulerCb` also acquires lock resources internally. My idea is that as long as `removeStoreLocked` is successful and all deletions are completed (last=true), then `removeSchedulerCb` should be executed. However, the logic here seems a bit confusing, making it difficult to modify. It may be necessary to reorganize and refactor this part. But it's not related to this PR.